servicetalk-http-netty/src/testFixtures/java/io/servicetalk/http/netty/AsyncContextHttpFilterVerifier.java

```diff
@@ -0,0 +1,205 @@
+                            assertAsyncContext(K2, V2, errorQueue);
+                            assertAsyncContext(K3, V3, errorQueue);
+                            assertAsyncContext(K4, V4, errorQueue);
+                        }
+
+                        @Override
+                        public void onError(final Throwable t) {
+                            assertAsyncContext(K1, V1, errorQueue);
+                            assertAsyncContext(K2, V2, errorQueue);
+                            assertAsyncContext(K3, V3, errorQueue);
+                            assertAsyncContext(K4, V4, errorQueue);
+                        }
+
+                        @Override
+                        public void onComplete() {
+                            assertAsyncContext(K1, V1, errorQueue);
+                            assertAsyncContext(K2, V2, errorQueue);
+                            assertAsyncContext(K3, V3, errorQueue);
+                            assertAsyncContext(K4, V4, errorQueue);
+                        }
+                    })
+                );
+            });
+        };
+    }
+
+    private static <T> void assertAsyncContext(final AsyncContextMap.Key<T> key, final T expectedValue,
+                                               final Queue<Throwable> errorQueue) {
+        final T actualValue = AsyncContext.get(key);
+        if (!expectedValue.equals(actualValue)) {
+            AssertionError e = new AssertionError("unexpected value for " + key + ": " +
+                    actualValue + " expected: " + expectedValue);
+            errorQueue.add(e);
+        }
+    }
+
+    private static void assertEmpty(Queue<Throwable> errorQueue) {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try (PrintStream ps = new PrintStream(baos, true, UTF_8.name())) {
+            Throwable t;
+            while ((t = errorQueue.poll()) != null) {
+                t.printStackTrace(ps);
+                ps.println(' ');
+            }
+            String data = new String(baos.toByteArray(), 0, baos.size(), UTF_8);
+            if (!data.isEmpty()) {
+                throw new AssertionError(data);
+            }
+        } catch (UnsupportedEncodingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static StreamingHttpServiceFilterFactory asyncContextAssertionFilter(
+            final BlockingQueue<Throwable> errorQueue) {
+        return service -> new StreamingHttpServiceFilter(service) {
+            @Override
+            public Single<StreamingHttpResponse> handle(final HttpServiceContext ctx,
+                                                        final StreamingHttpRequest request,
+                                                        final StreamingHttpResponseFactory responseFactory) {
+                return super.handle(ctx, request, responseFactory).liftSync(
+                        new BeforeFinallyHttpOperator(new TerminalSignalConsumer() {
```

Comment:
Because your assertions are identical, consider using `BeforeFinallyHttpOperator(Runnable)` overload which will invoke the same `Runnable` in all 3 cases.